web/src/drafts.ts

```diff
@@ -71,21 +71,103 @@ const draft_schema = z.intersection(
     ]),
 );
 
+const possibly_buggy_draft_schema = z
+    .intersection(
+        z.object({
+            content: z.string(),
+            updatedAt: z.number(),
+        }),
+        z.union([
+            z.object({
+                type: z.literal("stream"),
+                topic: z.string().optional(),
+                stream_id: z.number().optional(),
+                stream: z.string().optional(),
+            }),
+            z.object({
+                type: z.literal("private"),
+                reply_to: z.string(),
+                private_message_recipient: z.string(),
+            }),
+        ]),
+    )
+    .optional()
+    .catch(undefined);
```

Comment:
I'm assuming we don't want to do this for the regular `draft_schema` parsing, because we don't expect to be creating buggy drafts from now onwards. Does that sound correct, or should I add more safeguarding in case that did happen?

Reply:
Correct, with `TypeScript` it should be hard to create a new invalid draft.

Reply:
Hmm, looking at https://zod.dev/?id=catch, maybe we should pass a function to log a blueslip error about any parse failures, since we expect them to be impossible? Seems cheap and helpful.

Reply:
I just removed the `.optional.catch(undefined)` and it will throw zod parse exception if encounters drafts that it can't parse. This will be pretty disruptive, because then none of the drafts will be able to get fetched, but if we test enough on CZO then hopefully it'll be fine, and at least we won't be accidentally deleting drafts.

Reply:
Would you prefer we use `catch` specifically to use `blueslip`? Or is `zod` throwing an error just as fine.

Also, I guess I should be clear that up until now I did have a blueslip error here:

```js
        for (const [draft_id, draft] of Object.entries(parsed_drafts)) {
            // This is broken in a way we don't understand. It will get removed.
            if (draft === undefined) {
                blueslip.error("Found invalidly formatted draft", {draft_id});
                continue;
            }
```

But I'm removing that now that we're not generating `undefined` drafts.

Reply:
I think the ideal behavior for drafts that fail to parse would be to log a `blueslip.error` rather than throwing an exception so one can access drafts that don't have that problem, but not delete the draft from the store, which would (A) be bad and (B) make it impossible to debug what the story was with the draft that failed.

For chat.zulip.org deployment, throwing an exception is a fine way to achieve that.